src/script/CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+# Must be included before CMAKE_INSTALL_INCLUDEDIR is used.
+include(GNUInstallDirs)
+
+add_library(bitcoinconsensus
+  ../support/cleanse.cpp
+  bitcoinconsensus.cpp
```

Comment:
Very nice to see this properly handled now, even if it is a deviation from the current behavior.
Previously, `bitcoinconsensus.cpp` was always compiled and linked into most of the shipped binaries, even if it was not needed. The tests would also not actually link with `libbitcoinconsensus` to test the consensus code, but use the always available internal, static consensus library.
Now, it is only compiled for the consensus library, and the tests can correctly link with the external `libbitcoinconsensus`.